node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Reports.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.provision.node;
+
+import com.yahoo.slime.Cursor;
+import com.yahoo.slime.Inspector;
+import com.yahoo.slime.ObjectTraverser;
+import com.yahoo.slime.Type;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+
+/**
+ * @author hakonhall
+ */
+// @Immutable
+public class Reports {
+    private final Map<String, Report> reports;
+
+    public Reports() { this(new TreeMap<>()); }
+    private Reports(Map<String, Report> reports) { this.reports = Collections.unmodifiableMap(reports); }
```

Comment:
Should be `Maps.copyOf` to guarantee immutability, otherwise the caller can still modify the contents of `reports` through its own reference.

Reply:
Yes, but the caller is guaranteed to be the builder, so I thought we'd avoid an additional copy. Also, the Builder uses a TreeMap to guarantee a nice ordering of the elements.